core/dbt/deps/local.py

```diff
@@ -67,7 +67,13 @@ def install(self, project, renderer):
             system.make_symlink(src_path, dest_path)
         except OSError:
             fire_event(DepsSymlinkNotAvailable())
-            shutil.copytree(src_path, dest_path)
+            shutil.copytree(
+                src_path,
+                dest_path,
+                ignore=lambda directory, contents: project.packages_install_path
+                if directory == project.project_root
+                else [],
+            )
```

Comment:
### What

"Don't install yourself into your own [`dbt_packages`](https://docs.getdbt.com/reference/project-configs/packages-install-path) folder."

### How

[`shutil.copytree`](https://docs.python.org/3/library/shutil.html#shutil.copytree) recursively copies an entire directory tree rooted at `src` to a directory named `dst`.

The new `ignore` parameter introduces a base case under which this recursion should stop. It essentially says:

"No dbt project ever needs to install itself within it's own `packages_install_path` folder 😄. So when the current directory being processed is the same as _this_ dbt project, then do **not** recursively copy the configured [`packages_install_path`](https://docs.getdbt.com/reference/project-configs/packages-install-path) directory because it's already covered. Just ignore it!"

### Why

The `ignore` parameter is being added here to prevent a dbt project from installing itself via `dbt deps`. As mentioned in https://github.com/dbt-labs/dbt-core/issues/9719, this can happen if:
1. a dbt project is a subfolder of a [`local` dependency listed within `packages.yml`](https://docs.getdbt.com/docs/build/packages#local-packages), and
2.  the host operating system doesn't support symlinks.

The 1st item is common for CI testing for packages like `dbt_utils`, and the 2nd is most common in legacy versions of Windows.

Reply:
Windows 10 users can possibly avoid this code path by turning on Developer Mode (which may allow creating Symbolic Links without requiring admin privileges) [[1]](https://stackoverflow.com/questions/58038683/allow-mklink-for-a-non-admin-user/58048888#58048888).

But this PR is specifically intended to cover the case where the user's system doesn't allow symlinks for whatever reason. So Developer Mode for Windows 10 is not a sufficient workaround for all cases.